File: 백준/Silver/1149. RGB거리/RGB거리.cc
#include <cstdio>
#include <algorithm>
using namespace std;

int main() {
    int n;
    int result;
    int R[1001] = { 0 };
    int G[1001] = { 0 };
    int B[1001] = { 0 };


    scanf("%d", &n);

    for (int i = 1; i <= n; i++) {
        int cost[3];

        for (int j = 0; j < 3; j++)
            scanf("%d", cost + j);

        R[i] = min(G[i - 1], B[i - 1]) + cost[0];
        G[i] = min(R[i - 1], B[i - 1]) + cost[1];
        B[i] = min(R[i - 1], G[i - 1]) + cost[2];
    }
    result = min(R[n], min(G[n], B[n]));

    printf("%d\n", result);

    return 0;
}

File: 백준/Silver/1149. RGB거리/RGB거리.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

class Main {

	static int n;
	static int[][] cost;
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		n = Integer.parseInt(br.readLine());
		cost = new int[n][3];
		
		for(int i=0; i<n; i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			
			for(int j=0; j<3; j++) {
				cost[i][j] = Integer.parseInt(st.nextToken());
			}
		}
		
		int[][] dp = new int[n][3];
		
		for(int i=0; i<3; i++) {
			dp[0][i] = cost[0][i];
		}
		
		for(int i=1; i<n; i++) {
			for(int j=0; j<3; j++) {
				dp[i][j] = cost[i][j] + Math.min(dp[i-1][(j+1)%3], dp[i-1][(j+2)%3]);
			}
		}
		
		System.out.println(Math.min(dp[n-1][0], Math.min(dp[n-1][1], dp[n-1][2])));
	}

}
